lib/dlopen.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const binding = process.binding('dlopen');
+
+exports.dlopen = function dlopen(name) {
+  var lib = new Buffer(binding.sizeof_uv_lib_t);
```

Comment:
I'm concerned with making sure the new Buffer implementation meets all your needs. When Buffer moves to Uint8Array it will not have a weak callback available. Anyone who wishes to use one will have to externalize the object and setup the weak callback on their own.

I'm trying to recall a conversation we had 3+ years ago when I was doing my first rewrite of Buffer. I vaguely recall that for certain cases you wanted access to the weak callback because the lifetime of the pointer might have been different than that of the Object.